sdk/src/validate.ts

```
@@ -0,0 +1,82 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+const EInvalidYears = 'Coupon is not valid for the given number of years.';
+const EInvalidForNameLength = 'Coupon is not valid for the given name length.';
+const EInvalidPercentage = 'Invalid percentage amount for coupon.';
+const EInvalidUser = 'Coupon address does not match.';
+const ECouponExpired = 'Coupon has expired.';
+const EInvalidAvailableClaims = 'Number of claims cannot be zero.';
+
+export function hasAvailableClaims(rules: { available_claims?: string | null }): boolean {
+    if (rules.available_claims === null || rules.available_claims === undefined) {
+        return true;
+    }
+    if (parseInt(rules.available_claims) > 0) {
+        return true;
+    }
+    throw new Error(EInvalidAvailableClaims);
+}
+
+export function isCouponValidForNameYears(
+    rules: { years?: { from: number; to: number } | null },
+    years: number,
+): boolean {
```

I see no point to return a boolean if it's always going to be `true`.
I suggest it be a `void` return type and just let it throw the error if validation fails.